src/App.tsx

```diff
@@ -1,12 +1,62 @@
 import React from 'react';
 import './App.scss';
+import GoodsList from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+const goodsFromAPI = () => {
+  return fetch('https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json')
+    .then(res => res.json());
+};
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[];
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+  };
+
+  showAll = () => {
+    goodsFromAPI()
+      .then(goods => {
+        this.setState({
+          goods,
+        });
+      });
+  };
+
+  show5 = () => {
+    goodsFromAPI()
+      .then(goods => {
+        const filteredFive = goods.filter((good: { id: number; }) => good.id <= 5);
+
+        this.setState({
+          goods: filteredFive,
+        });
+      });
+  };
+
+  showRed = () => {
+    goodsFromAPI()
+      .then(goods => {
+        const filteredRed = goods.filter((good: { color: string; }) => good.color === 'red');
+
+        this.setState({
+          goods: filteredRed,
+        });
+      });
+  };
+
+  render() {
+    return (
+      <div className="main">
+        <button type="button" onClick={this.showAll} className="btn btn-outline-primary button">Show All</button>
```

Comment:
переноси атрибуты, если их больше двух
```suggestion
        <button
          type="button"
          onClick={this.showAll}
          className="btn btn-outline-primary button"
        >
          Show All
        </button>
```